### PR TITLE
Update NPCUseActionOnTargetSystem

### PR DESCRIPTION
**Pull request by comasqw**
## About the PR

It is now possible to specify multiple actions in NPCUseActionOnTargetSystem.
You can also specify the chance of execution for each action.
## Technical details

before - 
image
after -  
image

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [ ] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

Now the "NPCUseActionOnTarget" component receives a dictionary called "actions", where the key is the action ID, and the value is the probability of that action being performed.

**Changelog**

### CODE REVIEW

File: Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs [hunk 1]
```diff
@@ -20,15 +24,28 @@ public override void Initialize()
...
-        ent.Comp.ActionEnt = _actions.AddAction(ent, ent.Comp.ActionId);
+        foreach(var action in ent.Comp.Actions)
+        {
+            var key = action.Key;
+            var value = action.Value;
+
+            var actionEnt = _actions.AddAction(ent, key);
+
+            if (actionEnt is null)
+                return;
```
Comment (TheShuEd) [inline Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs:35]:
```suggestion
                continue;
```

File: Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs [hunk 2]
```diff
@@ -60,7 +77,26 @@ public override void Update(float frameTime)
...
+    {
+        List<EntityUid> keys = actions.Keys.ToList();
+        List<float> values = actions.Values.ToList();
+        EntityUid defaultToReturn = keys[values.IndexOf(values.Max())];
+
+        for (int i = 0; i < actions.Count; i++)
+        {
+            float maxProb = values.Max();
+            int maxindex = values.IndexOf(maxProb);
+            if (_random.NextFloat() <= maxProb)
```
Comment (TheShuEd) [inline Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs:94]: _random.Prob

File: Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs [hunk 3]
```diff
@@ -60,7 +77,26 @@ public override void Update(float frameTime)
...
+    }
+
+    private EntityUid ChooseAction(Dictionary<EntityUid, float> actions)
+    {
+        List<EntityUid> keys = actions.Keys.ToList();
+        List<float> values = actions.Values.ToList();
+        EntityUid defaultToReturn = keys[values.IndexOf(values.Max())];
+
+        for (int i = 0; i < actions.Count; i++)
+        {
```
Comment (TheShuEd) [inline Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs:91]: WeightRandom system, which is already implemented in EntityTable (I'm not sure), will fit here
see if you can't reuse it

Comment (comasqw) [reply Content.Server/NPC/Systems/NPCUseActionOnTargetSystem.cs:91]: > EntityTable

Not in EntityTable

**Comment by comasqw** [comment]
Also need to add a cooldown between the TryUseAction call, because there may be situations when a mob uses several different actions almost simultaneously.

**Review by metalgearsloth — COMMENTED** [review]
Chance of doing action A or action B should be handled in NPC code, not the part that interacts with the game.

You've also baked cooldowns in when this should also be handled via action cooldown or NPC behaviours.

**Pull request closed without merging**